kfs-core/src/main/java/edu/arizona/kfs/fp/document/GeneralErrorCorrectionDocument.java

```diff
@@ -51,5 +72,116 @@ public void toCopy() throws WorkflowException {
     	errorCertification.setErrorDescription(oldErrorCertification.getErrorDescription());
     	errorCertification.setErrorCorrectionReason(oldErrorCertification.getErrorCorrectionReason());
     }
-    
+
+
+    /**
+     * The map key is a combination of the accounting line sequence number and type (e.g. 1F)
+     *
+     * @return
+     */
+    public Map<String, Entry> getEntries() {
+        // populate Entries on document load
+        Map<String, Entry> entries = new TreeMap<String, Entry>();
+        if (getGecEntryRelationships() != null && !getGecEntryRelationships().isEmpty()) {
+            for (GecEntryRelationship rel : getGecEntryRelationships()) {
+                entries.put(rel.getGecAcctLineSeqNumber() + rel.getGecFdocLineTypeCode(), rel.getEntry());
+            }
+        }
+
+        return entries;
+    }
+
+
+    public Entry getEntryByAccountingLine(AccountingLine line) {
+        if (line != null) {
+            return getEntries().get(line.getSequenceNumber() + line.getFinancialDocumentLineTypeCode());

```

Comment:
There's potential for the lookup/actions to swap out GecEntryRelationship members, but at that time constructs with an in-memory Entry, and we don't incur a hit at that time. Also, the GecEntryRelationship collection on this doc is registered with the delete aware list of super, so nulling out will have repercussions -- we switch on if Entry.gecDocNum is non-null or not, so need to make sure auto-nulling isn't happening. Note too, that entryId is an indexed GUID, so is quite snappy in search anyway. Lastly, this specific impl detail is mimicked from the UCI code.